### ISSUE TITLE
Champion "deconstruction in from and let"

### ISSUE THREAD

**Issue by gafter** [dotnet/csharplang#189]
- [ ] Proposal added
- [ ] Discussed in LDM
- [ ] Decision in LDM
- [ ] Finalized (done, rejected, inactive)
- [ ] Spec'ed

See also https://github.com/dotnet/roslyn/issues/13964

**Comment by 333fred** [comment]
> Is there any progress on this?

This feature is in the Any Time bucket, which means we are unlikely to spend time on it ourselves, but if a motivated community member would like to contribute, we would be happy to review. Currently, this issue needs a specification for LDM to review and approve.

**Comment by sab39** [comment]
Does this cover things like
```cs
from person in People
where person.Age is int age && age >= 18 // person.Age is a nullable int
select (person, age)
```
or would that be a separate ticket?

**Comment by 333fred** [comment]
I don't think that would be covered by this. That would be changing the scoping of existing code, which would be concerning.

**Comment by sab39** [comment]
Oof. I see what you mean - my code example would compile today if something called `age` existed in an outer scope. That seems to me like a footgun in the status quo, though: if someone declares a variable in a pattern expression in a `where` clause and then uses the same identifier in the next line in a `select` clause, it's unlikely that they _intended_ to pull in a different value from outside, and even more unlikely that someone reading the code would recognize what's happening.

Of course, breaking existing code is a big deal and should only be done in extenuating circumstances. But does "the code is probably not doing what they intended in the first place" count as an extenuating circumstance?
